app/models/payment_provider_customers/base_customer.rb

```diff
@@ -35,6 +35,10 @@ def provider_payment_methods
       nil
     end
 
+    def service
```

Comment:
Not a huge fan of this. `service` sounds really generic, and somehow breaks the single responsibility principle. What do you think of using the factory pattern (similar to the `BillableMetrics::AggregationFactory`) to initialize the right service based on the customer?

Reply:
so having a `service_class` on the model?

Reply:
I agree with @vincent-pochet that model shouldn't know anything about the service (or a service class).